cms/djangoapps/maintenance/views.py

```diff
@@ -0,0 +1,231 @@
+    @method_decorator(require_global_staff)
+    def post(self, request):
+        """
+        Force publishes a course.
+
+        Arguments:
+            course_id (string): a request parameter containing course id
+            is_dry_run (string): a request parameter containing dry run value.
+                                 It is obtained from checkbox so it has either values 'on' or ''.
+        """
+
+        course_id = request.POST.get('course-id')
+        is_dry_run = bool(request.POST.get('dry-run'))
+
+        self.context.update({
+            'form_data': {
+                'course_id': course_id,
+                'is_dry_run': is_dry_run
+            }
+        })
+
+        try:
+            course_usage_key = self.validate_course_key(course_id)
+        except InvalidKeyError:
+            self.context['error'] = True
+            self.context['msg'] = COURSE_KEY_ERROR_MESSAGES['invalid_course_key']
+        except ItemNotFoundError as exc:
+            self.context['error'] = True
+            self.context['msg'] = exc.message
+        except ValidationError as exc:
+            self.context['error'] = True
+            self.context['msg'] = exc.message
+
+        if self.context['error']:
+            return self.render_response()
+
+        source_store = modulestore()._get_modulestore_for_courselike(course_usage_key)  # pylint: disable=protected-access
+        if not hasattr(source_store, 'force_publish_course'):
+            self.context['msg'] = _("Force publish course does not support old mongo style courses.")
+            logging.info(
+                "Force publish course does not support old mongo style courses. \
+                %s attempted to force publish the course %s.",
+                request.user,
+                course_id,
+                exc_info=True
+            )
+            return self.render_response()
+
+        current_versions = get_course_versions(course_id)
+
+        # if publish and draft are NOT different
+        if current_versions['published-branch'] == current_versions['draft-branch']:
+            self.context['msg'] = _("Course is already in published state.")
+            logging.info(
+                "Course is already in published state. %s attempted to force publish the course %s.",
+                request.user,
+                course_id,
+                exc_info=True
+            )
+            return self.render_response()
+
+        self.context['current_versions'] = current_versions
+
+        if is_dry_run:
+            logging.info(
+                "%s dry ran force publish the course %s.",
+                request.user,
+                course_id,
+                exc_info=True
+            )
+            return self.render_response()
+        updated_versions = source_store.force_publish_course(
+            course_usage_key, request.user, commit=True
+        )
+        if not updated_versions:
+            self.context['msg'] = _("Could not publish course.")
```

Comment:
Now that I have gone deeper into this, I think, this message and check won't be necessary. Because, the only way `updated_versions` is None is when we have faulty course locator which we have already taken care in the very start of the post method.